modules/codegen/src/smithy4s/codegen/internals/SmithyToIR.scala

```diff
@@ -683,6 +683,28 @@ private[codegen] class SmithyToIR(model: Model, namespace: String) {
 
     }
 
+  private def imputeZeroValuesOnDefaultTraits(
+      shape: Shape
+  ): List[Trait] = shape.getAllTraits().asScala.values.toList.map {
+    case default: DefaultTrait if default.toNode == Node.nullNode =>
```

Comment:
got to ask the obvious question, does the dynamic module handle this? or would things just blow up?

Reply:
It won't blow up, because [this](https://github.com/disneystreaming/smithy4s/blob/ba331d19fc45db1aba3331a9997776393a00bb05/modules/core/src/smithy4s/schema/Schema.scala#L84), but it won't be giving the same behaviour either. 

However, @kubukoz does have a point. This should also be handled at runtime in that `getDefaultValue` method.

Reply:
Just to clarify, you mean handled at runtime instead of changing the generated code? Or in addition to?

Reply:
In addition to. The generated code bit is required for default parameters of scala constructors and methods. The runtime bit is required for interpreters applied on dynamic service instances.

Reply:
For implementing the `getDefaultValue` method, I will need to know the type of schema I am working with so I know which value to transform the null default to. However, I don't think a simple pattern match on the current schema will work due to lazy schema.. do I need to implement a simple schema visitor to accomplish this? Or do you have another recommendation?

Reply:
Yeah I think a simple schema visitor might actually be a good idea.